lib/Twig/Extensions/SimpleTokenParser.php

```diff
@@ -19,6 +23,8 @@
      */
     public function parse(Twig_Token $token)
     {
+        @trigger_error('The grammar feature is deprecated since version 1.5 and will be removed in 2.0.', E_USER_DEPRECATED);
```

Comment:
this should be triggered when autoloading the class. Here, if you register a grammar-based token parser but don't use the associated tag in the template, you never get the deprecation (but you still get the fatal error when the class gets removed).

And if you decide to keep the warning here instead, you should mention the tag name in the deprecation to make it easier to know which tag uses the feature (the stack trace may not help as it will go in the Twig parser).